lib/elixir/lib/enum.ex

```diff
@@ -717,6 +717,32 @@ defmodule Enum do
   """
   @spec fetch(t, integer) :: {:ok, element} | :error
   def fetch(enumerable, index)
+  def fetch(first.._, 0),
```

Comment:
Still not a big fun of such small clauses specially given the common case should be fast enough.